### DISCUSSION TITLE
Interpretation of the end cue

### DISCUSSION THREAD

**Discussion by chenlung** [SubtitleEdit/subtitleedit#6650] [General]
[Rather than file directly to the issue tracker, I thought it best to discuss first.]

The start cue of a subtitle seems unambiguous; it should display it. However, the end cue has been open to interpretation across software; should it show or hide on the denoted frame or milliseconds?

Currently, Subtitle Edit (3.6.11) shows the frame. This approach seems one also adopted by Annotation Edit.
![Interpretation of the end cue  Subtitle Edit](https://user-images.githubusercontent.com/2973439/216789103-31285f62-4def-48a9-a9bf-ea4790631c67.jpg)

However, some others ([EZTitles](https://www.eztitles.com), I believe; [YouTube](https://www.youtube.com); [Vimeo](https://vimeo.com); the upcoming version of [Subtitle Horse](https://subtitle-horse.com)) hide on it; I think Subtitle Edit should adopt this approach (if a standard exists and how major organisations do it).

[Editing Tools](https://en.editingtools.io/subtitles) introduced a function (requested by me, incidentally) to shorten or extend by one frame. Perhaps Subtitle Edit should introduce a similar function for legacy files or importing from applications that export with the same approach as SE currently.

[Here](https://github.com/SubtitleEdit/subtitleedit/files/10609639/Interpretation.of.the.end.cue.zip) is material for testing.

**Comment by niksedk** [comment]
The "End time" in SE should probably be "Hide time" or "Out time" - any preferred phrase? @LeonCheung @OmrSi  ?
SE just started with "Start time" and "End time" which seem logical, but it's probably not well chosen...

**Comment by OmrSi** [comment]
As far as I know, Netflix and EZ call them `in-cue` and `out-cue`.
But I've never had an issue with `Start time`.

As for the main subject of discussion, most programs and platforms hide on the out-cue/end time, and I think this is the correct way because it's consistent with how the video players show it and with the rules of shot changes.

**Comment by chenlung** [comment]
How do you propose we approach the issue?

If the change is made to _hide_ on the end cue, I would suggest the following:
- Advertise it (it will be important for others)
- A function to add a frame's worth on the end cue for 'legacy' projects/files

**Comment by chenlung** [comment]
#3897

**Comment by chenlung** [comment]
The test suite [here](https://github.com/SubtitleEdit/subtitleedit/issues/7784#issuecomment-1868507791) should help.
